scripts/globals/item_utils.lua

```diff
@@ -41,3 +41,28 @@ end
 xi.item_utils.skillBookUse = function(target, skillID)
     target:trySkillUp(skillID, target:getMainLvl(), true, true)
 end
+
+xi.item_utils.pickItemRandom = function(target, itemgroup) -- selects an item from a weighted result table
```

Comment:
_Shockingly_ we didn't already have this. Will likely expand this out to multiple other places where we basically do the exact same thing.